lib/yubihsm_winhttp.c

```diff
@@ -309,7 +311,10 @@ static yh_rc backend_send_msg(yh_backend *connection, Msg *msg, Msg *response,
   }
 
   if (identifier != NULL && strlen(identifier) > 0 && strlen(identifier) < 32) {
-    swprintf(hsm_identifier, 128, L"YubiHSM-Session: %s", identifier);
+    tmp_identifier_len = mbstowcs(NULL, identifier, 0);
+    tmp_identifier = calloc(tmp_identifier_len + 1, sizeof(wchar_t));
+    mbstowcs(tmp_identifier, identifier, tmp_identifier_len + 1);
+    swprintf(hsm_identifier, 128, L"YubiHSM-Session: %s", tmp_identifier);
```

Comment:
Did you try this with %S or %ls instead of doing the conversion by hand? Since we know this code now only compiles correctly with MSVC should we use #error (or #warn) if this path is hit with gcc?

Reply:
Ahh... sorry, I forgot about this. Fixed how